### PR TITLE
MULE-11017 use schemas to validate xml files

### PR DESCRIPTION
**Pull request by fernandezlautaro**
Before this PR Mule wasn't properly validating the XML files through XSD. What the `XmlConfigurationDocumentLoader` will do is gather ALL errors and display them in the following format when there was 1 error found:
```
org.mule.runtime.core.api.config.ConfigurationException: org.mule.runtime.api.exception.MuleRuntimeException: There was '1' error while parsing the file 'update-name-param-override-config.xml'.
Full list:
org.xml.sax.SAXParseException; lineNumber: 10; columnNumber: 30; cvc-complex-type.2.4.a: Invalid content was found starting with element 'db:input-parameters'. One of '{"http://www.mulesoft.org/schema/mule/db":parameter-types}' is expected.
 (org.mule.runtime.api.exception.MuleRuntimeException)... bla bla bla full stack below
```
or the following when there are 2 or more:
```
org.mule.runtime.core.api.config.ConfigurationException: 
org.mule.runtime.api.exception.MuleRuntimeException: java.lang.IllegalArgumentException: There are '2' errors while parsing the file 'some-xml-filename.xml'.
Full list:
org.xml.sax.SAXParseException; lineNumber: 17; columnNumber: 97; cvc-complex-type.3.2.2: Attribute 'key' is not allowed to appear in element 'httpn:request'.
org.xml.sax.SAXParseException; lineNumber: 21; columnNumber: 136; cvc-complex-type.3.2.2: Attribute 'key' is not allowed to appear in element 'httpn:request'.
 (org.mule.runtime.api.exception.MuleRuntimeException)... bla bla bla full stack below
```

CB build for CE+EE: https://muleesb.ci.cloudbees.com/view/Template%20Jobs/job/CE-EE-Test-4.x/105/ , is almost green =].
It fails the same tests of ToolingServiceTestCase of the EE system tests (consistent on what the current SNAPSHOT build is) com.mulesoft.mule.distributions.server.tooling.ToolingServiceTestCase.getMetadataKeysThroughSer
and com.mulesoft.mule.distributions.server.tooling.ToolingServiceTestCase.getMetadataThroughService

### CODE REVIEW

File: modules/spring-config/src/main/java/org/mule/runtime/config/spring/MuleArtifactContext.java [hunk 1]
```diff
@@ -206,7 +206,8 @@ private void createApplicationModel() {
       ModuleExtensionStore moduleExtensionStore = new ModuleExtensionStore();
       for (Resource springResource : artifactConfigResources) {
         Document document =
-            xmlConfigurationDocumentLoader.loadDocument(Optional.of(moduleExtensionStore), springResource.getInputStream());
+            xmlConfigurationDocumentLoader.loadDocument(Optional.of(moduleExtensionStore), springResource.getFilename(),
```
Comment (estebanwasinger) [inline modules/spring-config/src/main/java/org/mule/runtime/config/spring/MuleArtifactContext.java]: `Optional.of()` -> `of()`

**Comment by alegmarra** [comment]
👍  SDK part

**Comment by fernandezlautaro** [comment]
@pablolagreca I added a factory implementation to for the `XmlConfigurationDocumentLoader` as we talked about and some tests to guarantee that if we feed a custom implementation of `XmlGathererErrorHandler`, the parsing can be done retrieving a `org.w3c.dom.Document` even though it has errors in it.

File: modules/spring-config/src/test/java/org/mule/runtime/config/spring/XmlConfigurationDocumentLoaderTestCase.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+import javax.xml.xpath.XPathExpressionException;
+
+import org.hamcrest.core.StringContains;
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.xml.sax.SAXException;
+import org.xml.sax.SAXParseException;
+
+public class XmlConfigurationDocumentLoaderTestCase {
```
Comment (pablolagreca) [inline modules/spring-config/src/test/java/org/mule/runtime/config/spring/XmlConfigurationDocumentLoaderTestCase.java]: must extend AbstractMuleTestCase

**Pull request merged**